src/CodeGen_LLVM.cpp

```diff
@@ -2475,7 +2475,23 @@ void CodeGen_LLVM::visit(const Call *op) {
         internal_assert(op->args.size() == 3);
         Expr cond = op->args[0];
         if (cond.type().is_vector()) {
-            scalarize(op);
+            // This will dramatically slow down the code, so it's worthy
+            // of a warning to user, in case it was inserted in a non-obvious way.
+            user_warning << "The condition for a require() expr is vectorized; it will be split and scalarized.";
+            if (!cond.type().is_bool()) {
+                // If we're building for a backend that uses bool_to_mask(),
+                // the vector might no longer be booleans, and scalarizing
+                // can fail at codegen time (since the conditions are expected)
+                // to be boolean.
+                cond = (cond != make_zero(cond.type()));
```

Comment:
I think the right thing to do here might be to to make it so scalarize uses `Call::extract_mask_element`.

Reply:
AFACT `extract_mask_element` is only defined for the HVX backend. (Maybe a better fix is to specialize `require` for that backend...)

Reply:
That just looks like a bug. It looks like the implementation of extract_mask_element in CodeGen_Hexagon should be in CodeGen_LLVM.

Reply:
Hmm, it looks the same for bool_to_mask, but maybe there was a reason for this. I vaguely recall discussing this with @abadams...

Reply:
This is by no means an urgent fix, so I'll wait to see if @abadams chimes in with any recollection about it.

Reply:
Thought about this more. I think the mask intrinsic implementations should stay in the codegen backends, scalarize should use extract_mask_element, and it should be implemented on any target that uses eliminate_bool_vectors. Since OpenCL doesn't use CodeGen_LLVM, everything I said above wouldn't impact it anyways.

Reply:
Moved Call::extract_mask_element from Hexagon to LLVM backend.